envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -482,6 +484,24 @@ fun Value.toDuration(): Duration? {
     }
 }
 
+fun Value?.toCustomData(): Map<String, Any?> {
+    return when (this?.kindCase) {
+        Value.KindCase.STRUCT_VALUE -> this.toCustomDataValue() as Map<String, Any?>
```

Comment:
`Value?.toCustomDataValue()` returns `Any?`, so casting it to Map may result in an exception. I know that the current logic prevents it, but I think that this two methods are too high coupled. Could you refactor this?

Reply:
Please check it now